server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1188,13 +1188,22 @@ public Index(Engine.Index index, Engine.IndexResult indexResult) {
             this.autoGeneratedIdTimestamp = index.getAutoGeneratedIdTimestamp();
         }
 
+        // only for testing
         public Index(String id, long seqNo, long primaryTerm, byte[] source) {
```

Comment:
Thanks for your feedback David. I'm doing baby steps here in the context of #90498 where I'd like to randomize the XContentType of the source of translog operations, so it's likely that the string will end up being JSON/SMILE etc. But there will be cases in tests where having a method in the test framework will be useful so it's a good idea. I suggest to do this in #90498 (not ready for review yet), WDYT?

Reply:
Up to you really, but I think it might make #90498 a little easier to get rid of this test-only constructor now. The resulting factory method would be a good place to add some randomisation perhaps.